extending/sanitizers.rst

```diff
@@ -0,0 +1,167 @@
+.. _sanitizers:
+
+Compiler sanitizers
+===================
+
+Sanitizers are tools that can detect bugs such as buffer overflows or accesses, dangling pointer or
+different types of undefined behavior.
+
+The two compilers that mainly support sanitizing options are gcc and clang. These options are
+passed to the compiler as flags and  depending on if you are using
+`clang <https://clang.llvm.org/docs/UsersManual.html#controlling-code-generation>`_ or
+`gcc <https://gcc.gnu.org/onlinedocs/gcc/Instrumentation-Options.html>`_ different sanitizers are
+supported.
+
+Here we explain different options on how to model and use sanitizers with your Conan packages.
+
+Adding custom settings
+----------------------
+
+If you want to model the sanitizer options so that the package id is affected by them you have to
+introduce new settings in the *settings.yml* file (see :ref:`custom_settings` section for more
+information).
+
+Sanitizer options should be modeled as sub-settings of the compiler. Depending on how you want to
+combine the sanitizers you have two choices.
+
+Adding a list of commonly used options
+######################################
+
+If you have a fixed set of sanitizers or combinations of them that are the ones you usually set for
+your builds you can add the sanitizers as a list of options. An example for *apple-clang* would be
+like this:
+
+.. code-block:: yaml
+    :caption: *settings.yml*
+
+    apple-clang:
+        version: ["5.0", "5.1", "6.0", "6.1", "7.0", "7.3", "8.0", "8.1", 
+                  "9.0", "9.1", "10.0", "11.0"]
+        libcxx: [libstdc++, libc++]
+        cppstd: [None, 98, gnu98, 11, gnu11, 14, gnu14, 17, gnu17, 20, gnu20]
+        sanitizer: [None, Address, Thread, UndefinedBehavior, AddressUndefinedBehavior]
```

Comment:
please highlight this line in the code-block

Reply:
Done, thanks!